### PR TITLE
Upgrade Script Mattermost

### PR DESCRIPTION
**Pull request by JtheBAB**
Follow up for #2502

makes it easier with a new request.

### CODE REVIEW

**Comment by jasonblais** [comment]
@justinegeffen replying to your note from #2502:

> Hi @JtheBAB. We have a new Community-Managed Documentation https://docs.mattermost.com/guides/administrator.html#community-managed-documentation) section in the Admin Guide. If this is still an applicable solution, I propose adding it there. @jasonblais, what are your thoughts?

I'm open to it, but would like to include the same note at the top of the page stating that it's not maintained or officially supported by the Mattermost team.

Perhaps in addition to including in the community-maintained docs section we also add it to the forums where people can easily share feedback on it? We can link to this forum post as well.

**Comment by jasonblais** [comment]
@JtheBAB Thank you also for this contribution! What is the latest release version you used it for?

@justinegeffen For reviewer, I would propose either Paul or Sven to review instead of Carlos, since they are familiar with typical customer upgrade processes.

**Comment by JtheBAB** [comment]
I use it for every upgrade on two instances since i created the pull request. Never had a problem. And i am on the latest MM version

File: source/administration/upgrade-script.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
+Upgrading Mattermost Server with a script
+=========================================
+
+*Preparing the script*
+
+`Save the script <https://docs.mattermost.com/administration/upgrade_mattermost.sh>`__ to your mattermost server.
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:6]:
```suggestion
`Save the script <https://docs.mattermost.com/administration/upgrade_mattermost.sh>`__ to your Mattermost server.
```

File: source/administration/upgrade_mattermost.sh [hunk 2]
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env bash
+
+# Immedially exit if a command run from a loop, a pipeline or a compound
```
Comment (justinegeffen) [inline source/administration/upgrade_mattermost.sh:3]:
```suggestion
# Immediately exit if a command run from a loop, a pipeline or a compound
```

File: source/administration/upgrade-script.rst [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Upgrading Mattermost Server with a script
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:1]:
```suggestion
Upgrading Mattermost Server With a Script
```

File: source/administration/upgrade-script.rst [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+=========================================
+
```
Comment (justinegeffen) [inline source/administration/upgrade-script.rst:3]: Add the following here:

.. important:: This unofficial guide is maintained by the Mattermost community and this deployment configuration is not yet officially supported by Mattermost, Inc. Community testing, feedback, and improvements are welcome and greatly appreciated.

**Comment by justinegeffen** [comment]
@jasonblais, I'm +1 on the note and forums idea.

@JtheBAB, I've made two minor suggestions and I've added the banner copy as a suggestion as well. Would you be open to posting on the forum as well? :)

File: source/administration/upgrade-script.rst (hunk 3 above)
Comment (JtheBAB) [reply source/administration/upgrade-script.rst:1]: Why is with With?

Comment (justinegeffen) [reply source/administration/upgrade-script.rst:1]: Ah that was meant to be "with", apologies. I was comparing heading casing and didn't go back to rectify this one.

**Comment by JtheBAB** [comment]
Added your suggestion @justinegeffen

**Review by justinegeffen — APPROVED** [review]
Thanks, @JtheBAB, this works for me. :)

**Review by svelle — APPROVED** [review]
LGTM! 

Looking forward to finally get this into the docs ;)

**Comment by justinegeffen** [comment]
@saturninoabril, gentle ping on this. Would be awesome to merge it today. :)

**Review by saturninoabril — APPROVED** [review]
Thanks @JtheBAB, LGTM!

**Comment by amyblais** [comment]
@justinegeffen There are a few issues here:

1. For ``administration/upgrade-script.rst``, the code blocks are not showing up at https://docs.mattermost.com/administration/upgrade-script.html.
2. For ``administration/upgrade_mattermost.sh``, the page doesn't appear to exists.

**Comment by justinegeffen** [comment]
Thanks, @amyblais. I've fixed 1. For 2, I didn't realize that, thanks for catching it. 

I've submitted a PR for the code rendering. I'll open another PR to add the full script to that page.

**Pull request merged**